Dockerfile

```diff
@@ -38,11 +40,18 @@ RUN apk add --no-cache bash ca-certificates curl jq wget nano
 
 COPY --from=build /build/cloudflare-ext-authz-service /cloudflare-ext-authz-service/run
 
-RUN GRPC_HEALTH_PROBE_VERSION=v0.3.1 && \
-    wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
+ARG GRPC_HEALTH_PROBE_VERSION=v0.3.1
+RUN GRPC_HEALTH_PROBE_URL=https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux- && \
+    case "${TARGETARCH}" in \
+      'amd64') GRPC_HEALTH_PROBE_URL="${GRPC_HEALTH_PROBE_URL}amd64" ;; \
+      'arm64') GRPC_HEALTH_PROBE_URL="${GRPC_HEALTH_PROBE_URL}arm64" ;; \
+      'arm')   GRPC_HEALTH_PROBE_URL="${GRPC_HEALTH_PROBE_URL}arm" ;; \
+      *)       echo "Unsupported arch: ${TARGETARCH}" && exit 1 ;; \
+    esac && \
+    wget -qO/bin/grpc_health_probe ${GRPC_HEALTH_PROBE_URL} && \
```

Comment:
Can this not be simplified to just:

```sh
 wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-${TARGETARCH} 
```

Checking for supported arch can be a separate statement or we can just like the wget fail.

Reply:
got it